### PR TITLE
List all installed CSI drivers info from cluster CNFCERT24-item 1

### PR DESCRIPTION
**Pull request by wying3**
add listing 3rd party CSI drivers info to diagnostic test

### CODE REVIEW

**Comment by jun-rh** [comment]
Can you put a real example of the claim file section of this in a comment?

**Comment by wying3** [comment]
> Can you put a real example of the claim file section of this in a comment?

kind: CSIDriver
metadata:
  labels:
    app: controller.csi.trident.netapp.io
name: csi.trident.netapp.io
selfLink: /apis/storage.k8s.io/v1/csidrivers/csi.trident.netapp.io

**Review by HamadiSE — DISMISSED** [review]

File: test-network-function/diagnostic/suite.go [hunk 1]
```diff
@@ -112,6 +125,18 @@ type NodesHwInfo struct {
 	Worker NodeHwInfo
 }
 
+// CsiDriver stores csi driver JSON output from  `oc get csidirver -o json` command
+type CsiDriver struct {
+	Kind     string `json:"kind"`
```
Comment (jun-rh) [inline test-network-function/diagnostic/suite.go:130]: what is the point of this field if it's always the same value?

File: test-network-function/diagnostic/suite.go [hunk 2]
```diff
@@ -112,6 +125,18 @@ type NodesHwInfo struct {
...
+	Metadata struct {
+		Labels struct {
```
Comment (jun-rh) [inline test-network-function/diagnostic/suite.go:132]: Should this be an array or list?

File: test-network-function/diagnostic/suite.go (hunk 1 above)
Comment (wying3) [reply test-network-function/diagnostic/suite.go:130]: just want to specifiy in the claim file that this part is CSI Driver info

**Comment by jun-rh** [comment]
Took a look at the output Varun provided. There seems to be a lot more info that could be potentially useful, like version, group police etc. Why don't we just capture everything as {}interface type, similar to the node summary, instead of debating what fields to pick and creating structures for them?

File: test-network-function/diagnostic/suite.go (hunk 2 above)
Comment (wying3) [reply test-network-function/diagnostic/suite.go]: changed to map list

File: test-network-function/diagnostic/suite.go [hunk 3]
```diff
@@ -112,6 +125,18 @@ type NodesHwInfo struct {
...
+			App string `json:"app"`
```
[the remark this answers is not in the document]
Comment (wying3) [reply test-network-function/diagnostic/suite.go]: changed to new data  structure to include all labels

**Comment by wying3** [comment]
> Took a look at the output Varun provided. There seems to be a lot more info that could be potentially useful, like version, group police etc. Why don't we just capture everything as {}interface type, similar to the node summary, instead of debating what fields to pick and creating structures for them?

changed made with new commit

**Review by jun-rh — APPROVED** [review]

**Review by HamadiSE — APPROVED** [review]

**Pull request merged**